python/revng/tupletree/__init__.py

```diff
@@ -370,3 +388,374 @@ def init_check_kw_only(self, *args, **kwargs):
         original_init(self, *args, **kwargs)
 
     base_class.__init__ = init_check_kw_only  # type: ignore
+
+
+def construct_type(type_info: dict, data):
+    # if data not `str` or `dict` then its already constructed so we just return it.
+    if not isinstance(data, str) and not isinstance(data, dict):
+        return data
+
+    if type_info["ctor"] == "enum":
+        return type_info["possibleValues"](data)
+    elif type_info["ctor"] == "native":
+        return type_info["type"](data)
+    if type_info["ctor"] == "parse":
+        if type_info["type"].__name__ in ("TypeDefinition", "Type"):
+            return type_info["type"].from_dict(**data)
```

Comment:
This should not be here.
Remember, stuff in `tutpletree` is agnostic to the specific implementation (e.g. Model) so it should not contain **any** class from an implementation (`MetaAddress`, `TypeDefinition`, etc.)

Reply:
[Now](https://github.com/revng/revng/pull/398/commits/66de4905df354dcb61deeb2fd2ec1fd7b959fa92#diff-9e8dc3a9971b797416c6f2ef663881111783a16e41bfb39319a70733bef21590R393) I check if the data `type` is `dict`. Is this ok?